File: rb/features.rb
# MediawikiAction SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module MediawikiActionFeatures
  def self.make_feature(name)
    case name
    when "base"
      MediawikiActionBaseFeature.new
    when "ratelimit"
      MediawikiActionRatelimitFeature.new
    when "retry"
      MediawikiActionRetryFeature.new
    when "test"
      MediawikiActionTestFeature.new
    when "timeout"
      MediawikiActionTimeoutFeature.new
    else
      MediawikiActionBaseFeature.new
    end
  end
end

File: py/mediawikiaction_sdk/features.py
# MediawikiAction SDK feature factory

from mediawikiaction_sdk.feature.base_feature import MediawikiActionBaseFeature
from mediawikiaction_sdk.feature.test_feature import MediawikiActionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MediawikiActionBaseFeature(),
        "test": lambda: MediawikiActionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
